example/10_convnd_fwd_multiple_d_multiple_reduce/common.hpp

```diff
@@ -34,103 +34,57 @@ using I4 = ck::int4_t;
 using I8  = std::int8_t;
 using I32 = std::int32_t;
 
-template <typename Ret>
-struct type_function
+template <typename ALay, typename BLay, typename DELay, typename RLay>
+struct LayoutSetting
 {
-    using type = Ret;
+    using ALayout  = ALay;
+    using BLayout  = BLay;
+    using DELayout = DELay;
+    using RLayout  = RLay;
 };
 
-template <ck::index_t>
-struct ALayoutSelector;
-
-template <ck::index_t>
-struct BLayoutSelector;
-
-template <ck::index_t>
-struct DELayoutSelector;
-
-template <ck::index_t>
-struct RLayoutSelector;
+template <ck::index_t NDimSpatial>
+struct LayoutSettingSelector;
 
 namespace ctl = ck::tensor_layout::convolution;
 
 template <>
-struct ALayoutSelector<1> : type_function<ctl::GNWC>
-{
-};
-
-template <>
-struct BLayoutSelector<1> : type_function<ctl::GKXC>
-{
-};
-
-template <>
-struct DELayoutSelector<1> : type_function<ctl::GNWK>
-{
-};
-
-template <>
-struct RLayoutSelector<1> : type_function<ctl::GNW>
+struct LayoutSettingSelector<1> final : LayoutSetting<ctl::GNWC, ctl::GKXC, ctl::GNWK, ctl::GNW>
 {
 };
 
 template <>
-struct ALayoutSelector<2> : type_function<ctl::GNHWC>
+struct LayoutSettingSelector<2> final : LayoutSetting<ctl::GNHWC, ctl::GKYXC, ctl::GNHWK, ctl::GNHW>
 {
 };
 
 template <>
-struct BLayoutSelector<2> : type_function<ctl::GKYXC>
-{
-};
-
-template <>
-struct DELayoutSelector<2> : type_function<ctl::GNHWK>
-{
-};
-
-template <>
-struct RLayoutSelector<2> : type_function<ctl::GNHW>
-{
-};
-
-template <>
-struct ALayoutSelector<3> : type_function<ctl::GNDHWC>
-{
-};
-
-template <>
-struct BLayoutSelector<3> : type_function<ctl::GKZYXC>
-{
-};
-
-template <>
-struct DELayoutSelector<3> : type_function<ctl::GNDHWK>
-{
-};
-
-template <>
-struct RLayoutSelector<3> : type_function<ctl::GNDHW>
+struct LayoutSettingSelector<3> final
+    : LayoutSetting<ctl::GNDHWC, ctl::GKZYXC, ctl::GNDHWK, ctl::GNDHW>
 {
 };
 
 template <ck::index_t NDimSpatial>
-using ALayout = typename ALayoutSelector<NDimSpatial>::type;
+using ALayout = typename LayoutSettingSelector<NDimSpatial>::ALayout;
 
 template <ck::index_t NDimSpatial>
-using BLayout = typename BLayoutSelector<NDimSpatial>::type;
+using BLayout = typename LayoutSettingSelector<NDimSpatial>::BLayout;
 
 template <ck::index_t NDimSpatial>
-using DELayout = typename DELayoutSelector<NDimSpatial>::type;
+using DELayout = typename LayoutSettingSelector<NDimSpatial>::DELayout;
 
 template <ck::index_t NDimSpatial>
-using RLayout = typename RLayoutSelector<NDimSpatial>::type;
+using RLayout = typename LayoutSettingSelector<NDimSpatial>::RLayout;
 
 struct ExecutionConfig final
 {
     bool do_verification = true;
-    int init_method      = 1;
-    bool time_kernel     = false;
+#if defined(BUILD_INT4_EXAMPLE) || defined(BUILD_INT8_EXAMPLE)
+    int init_method = 1;
+#else
+    int init_method = 2;
+#endif
```

Comment:
You actually don't need to differentiate those two cases. If integral types would be used even with floating point initialization the final data are casted to requested type ie. you're filling int8 tensor using `FillUniformDistribution` - values will be casted to `int8` type.

Reply:
But does that mean I have to use different initialization value in this case (`int8` example) rather than **0.0** and **0.1**? For the `a`, `b` argument? 
```c++
ck::utils::FillUniformDistribution<ADataType>{a, b}(conv_input.begin(),
                                                    conv_input.end());
```

Reply:
Okay, it make sense. Then it is unnecessary to choose between initialization methods for different data types.